crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -31,6 +31,21 @@ pub(crate) async fn farm(
     }: FarmingArgs,
     best_block_number_check_interval: Duration,
 ) -> Result<(), anyhow::Error> {
+    match std::panic::catch_unwind(|| fdlimit::raise_fd_limit()) {
+        Ok(Some(limit)) => log::info!("Increase file limit from soft to hard (limit is {limit})"),
+        Ok(None) => log::debug!("Failed to increase file limit"),
+        Err(err) => {
+            let err = if let Some(err) = err.downcast_ref::<&str>() {
+                err
+            } else if let Some(err) = err.downcast_ref::<String>() {
+                err
+            } else {
+                "Failed to fetch error"
```

Comment:
This path should be unreachable, as internally `fdlimit` uses panic macro, which should return [either `&str` or `String`](https://doc.rust-lang.org/std/macro.panic.html).

Should I add a comment stating that?

Reply:
If it is actually unreachable, then we need `unreachable!()`?

Reply:
It might be broken in the future, so I decided to leave it like this.
Right now this function just uses `panic` macro and that's it. It should return `&str` or `String`.

Do you think I should just put `unreachable`?

Reply:
I don't have strong opinion, but `unreachable!()` with a comment seems the most appropriate.